### PR TITLE
[Yaml] Add flag to dump numeric key as string

### PR DESCRIPTION
**Pull request by alamirault**
#17594

### CODE REVIEW

File: components/yaml.rst [hunk 1]
```diff
@@ -418,6 +418,22 @@ you can dump them as ``~`` with the ``DUMP_NULL_AS_TILDE`` flag::
     $dumped = Yaml::dump(['foo' => null], 2, 4, Yaml::DUMP_NULL_AS_TILDE);
     // foo: ~
 
+Dumping Numeric Keys as string
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometime YAML keys must be always string. This component don't convert keys to string by default but you can
+make it convert numeric keys as string by using the ``DUMP_NUMERIC_KEY_AS_STRING`` flag::
```
Comment (alamirault) [inline components/yaml.rst]: We can improve wording I think

File: components/yaml.rst [hunk 2]
```diff
@@ -418,6 +418,22 @@ you can dump them as ``~`` with the ``DUMP_NULL_AS_TILDE`` flag::
     $dumped = Yaml::dump(['foo' => null], 2, 4, Yaml::DUMP_NULL_AS_TILDE);
     // foo: ~
 
+Dumping Numeric Keys as string
```
Comment (xabbuh) [inline components/yaml.rst]:
```suggestion
Dumping Numeric Keys as Strings
```

File: components/yaml.rst [hunk 3]
```diff
@@ -418,6 +418,22 @@ you can dump them as ``~`` with the ``DUMP_NULL_AS_TILDE`` flag::
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (xabbuh) [inline components/yaml.rst]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: components/yaml.rst (hunk 1 above)
Comment (xabbuh) [reply components/yaml.rst]: Maybe something like this:

> By default, digit-only array keys are dumped as integers. You can use the `DUMP_NUMERIC_KEY_AS_STRING` flag if you want to dump string-only keys::

Comment (alamirault) [reply components/yaml.rst]: I like it !

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thank you Antoine.

**Pull request merged**